smalltalksrc/VMMaker/StackToRegisterMappingCogit.class.st

```
@@ -4269,15 +4314,25 @@ StackToRegisterMappingCogit >> ssPop: n popSpilled: popSpilled [
 	self assert: (simStackPtr - n >= methodOrBlockNumTemps
 				or: [(needsFrame not and: [simStackPtr - n >= 0])]).
 
+	self simulationOnly: [
+		"For debug purposes, clean the stack when in simulation mode"
+		| toPop |
+		toPop := 0.
+		n timesRepeat: [
+			(self ssValue: toPop) type: nil.
+			toPop := toPop + 1.
+		]
+	].
```

Why this?  Is not hiding possible errors in production?

---

In the abstract stack, when you pop, it only moves the stack top pointer, leaving garbage in the stack.
So, when you inspect the abstract stack, you have a mix of valid and invalid entries.
This nils the unused/popped entries only to help with inspecting and debugging...

I did not find a bug related to this :) so I don't think that there is a problem in production...